docs/rst/recording/tutorials/basic_example.rst

```diff
@@ -2,9 +2,263 @@
+
+The private data members of the class defines the DDS Topic, DataTypeKind, DDS Topic type and DynamicType.
+
+.. literalinclude:: ../../../../resources/dds/TypeLookupService/TypeLookupServicePublisher.h
+    :language: C++
+    :lines: 139-146
+
+The next lines show the constructor of the ``TypeLookupServicePublisher`` class that implements the publisher.
+The publisher is created with the topic and data type to use.
+
+.. literalinclude:: ../../../../resources/dds/TypeLookupService/TypeLookupServicePublisher.cpp
+    :language: C++
+    :lines: 45-54
+
+Inside the ``TypeLookupServicePublisher`` constructor are defined the DomainParticipantQos.
+As the publisher act as a server of types, set the QoS in order to send this information.
+Set ``use_client`` to ``false`` and ``use_server`` to ``true``.
+
+.. literalinclude:: ../../../../resources/dds/TypeLookupService/TypeLookupServicePublisher.cpp
+    :language: C++
+    :lines: 58-62
+
+Next, we register the type in the participant:
+
+1. Generate the dynamic type through ``generate_helloworld_type_()`` explained below.
+2. Set the data type.
+3. Create the ``TypeSupport`` with the dynamic type previously created.
+4. Set the ``type`` to ``TypeInformation`` and not to ``TypeObject`` since we want to send the information of the type and not the object.
+
+* This answer `DDS-XTypes 1.2. <https://www.omg.org/spec/DDS-XTypes/1.2>`_
+
+.. literalinclude:: ../../../../resources/dds/TypeLookupService/TypeLookupServicePublisher.cpp
+    :language: C++
+    :lines: 73-95
+
+The function ``generate_helloworld_type_()`` returns the dynamic type generated with the object and id of the type.
+
+.. literalinclude:: ../../../../resources/dds/TypeLookupService/TypeLookupServicePublisher.cpp
+    :language: C++
+    :lines: 256-271
+
+Then are initialized the Publisher, DDS Topic and DDS DataWriter.
+
+To make the publication, the public member function ``publish()`` is implemented:
+
+1. It create the variable that will contain the user data, ``dynamic_data_``.
+2. Fill that variable with the function ``fill_helloworld_data_(msg)``, explained below.
+
+.. literalinclude:: ../../../../resources/dds/TypeLookupService/TypeLookupServicePublisher.cpp
+    :language: C++
+    :lines: 229-254
+
+The function ``fill_helloworld_data_()`` returns the data to be sent with the information filled in.
+
+First, it is created the ``Dynamic_ptr`` that will be filled in and return.
+With the use of ``DynamicDataFactory`` we create the data that corresponds to our data type.
+Finally, data variables are assigned, in this case, ``index`` and ``message``.
+
+.. literalinclude:: ../../../../resources/dds/TypeLookupService/TypeLookupServicePublisher.cpp
+    :language: C++
+    :lines: 293-307
+
+**************
+DDS Subscriber
+**************
+
+The DDS Subscriber is acting as a client of types.
+Thus, the subscriber does not need to know the type.
```

Comment:
```suggestion
The DDS Subscriber is acting as a client of types, i.e. the subscriber will not know the types beforehand and it will discovery the data type via the type lookup service implemented on the publisher side. 
```